src/queues.c

```diff
@@ -569,7 +569,9 @@ gint64 queues_get_next_datachange(gint64 time)
                 }
 
                 if (settings.show_age_threshold >= 0) {
-                        gint64 age = time - n->timestamp;
+                        /* When displaying age, floor the timestamp to the
+                         * second. See notification.c for details.*/
+                        gint64 age = time - (n->timestamp - (n->timestamp % S2US(1)));
```

Comment:
I'd suggest to move the flooring to inside the if statement. That's a little more explicit and makes sure you aren't too early.
```c
gint64 next_second = time - time % S2US(1);
sleep = MIN(sleep, (S2US(1) - (next_second % S2US(1))));
```

Reply:
Done in the current revision.